driver/src/orderbook/streamed/state.rs

```diff
@@ -73,11 +75,12 @@ impl State {
                 let token_id = self.tokens.get_id_by_address(*token_address)?;
                 Some((
                     (*user_id, token_id),
+                    // We fetch the balance including all trades/deposits that happened while collecting orders
+                    // for the batch we are trying to solve (balance at the beginning of the next batch).
                     // Can fail if user's balance exceeds U256::max.
-                    // Can fail while not all trades of a solution have been received and batch_id
-                    // is the next batch_id so that we assume that the current solution won't be be
-                    // reverted.
-                    bigint_u256::bigint_to_u256(&balance.get_balance(batch_id))?,
+                    bigint_u256::bigint_to_u256(
+                        &balance.get_balance_at_beginning_of_batch(batch_id + 1),
```

Comment:
Won't this include fluxes (like withdraw requests) that only mature in the next batch? Basically, is `end_of_batch(batch_id) <=> beginning_of_batch(batch_id + 1)`.

If I understand correctly, we do this to ensure that solutions and trades that just happened are included in the next batch?

Sorry, I should know this but I always get confused.